### ISSUE TITLE
Increasing vertex count on every frame causes crash

### ISSUE THREAD

**Issue by froody** [adrien-ben/imgui-rs-vulkan-renderer#30]
I was noticing a crash in my app where typing quickly (in an imgui window which displays the text) causes a DEVICE_LOST error on MacOS (using MoltenVK). I debugged this and it turned out to by a free-while-in-use of the vertex buffer. See [code](https://github.com/adrien-ben/imgui-rs-vulkan-renderer/blob/master/src/renderer/mod.rs#L750)

I got this message from the Metal debugger:
```
-[MTLDebugDevice notifyExternalReferencesNonZeroOnDealloc:]:2885: failed assertion `The following Metal object is being destroyed while still required to be alive by the command buffer 0x13180a600 (label: vkQueueSubmit CommandBuffer on Queue 0-0):
```

I'm not sure how to fix this, maybe with a counter in `mesh` to wait at least `in_flight_frames` before calling `allocator.destory_buffer`

**Comment by adrien-ben** [comment]
Hi, thanks for reporting and for the PR. 

I'll look at it today or tomorrow before pushing the 1.6.

**Comment by adrien-ben** [comment]
I released the 1.6 anyways and will push a 1.6.1 if needed.

I tried to replicate your behavior in an application using multiple frames in flight (the examples from this repo don't) but did not manage to.

If you use the same number of frames in flight in your application and in the renderer by passing an `Option` struct with `in_flight_frames` at creation time this should not happen. 
When you call draw you prepare the next frame on the CPU when the previous frame is being rendered. The previous frame uses a different buffer. If it's not the case, even when you're are not resizing the buffer and crashing you're updating the content of the buffer in use and could experience artifacts (i guess the actual behavior depends on the driver).

Have you tried running your app with validation layers to detect incorrect synchronization ?
